baggage/baggage.go

```diff
@@ -67,13 +67,27 @@ func NewKeyProperty(key string) (Property, error) {
 // NewKeyValueProperty returns a new Property for key with value.
 //
 // If key or value are invalid, an error will be returned.
+//
+// Deprecated: Use NewKeyValuePropertyRaw instead
+// that does not require precent-encoding of the value.
```

Comment:
I feel like we can leave this function un-deprecated. If we make it clear the difference between this and `NewKeyValuePropertyRaw` a user can decide if they want to use the added functionality this function provides or just skip it.

I think instead of deprecating, we should make this and `NewKeyValuePropertyRaw` clear about what different functionality they provide.

Reply:
I personally feel like it _should_ be deprecated because it just does an outright wrong thing that the users should never be using - it breaks a cross-language portability of baggage by introducing an encoding that does not belong at this level.

Reply:
I would prefer deprecating as well. 

I cannot come up with a scenario where the user would like to use  `NewMember` and `NewKeyValueProperty` that decodes the passed value. Users that would decode the baggage from headers would simply use `Parse` function which takes care of decoding. 

If people are already using  `NewMember` and `NewKeyValueProperty` then there is a high probablity that they are doing it wrong (that they are not precent-encoding the value) and thus they can experience bugs like https://github.com/open-telemetry/opentelemetry-go/issues/3685. The deprecation notice can help them migrating to the new API which would not lead of such issues.

Reply:
After some [thoughts](https://github.com/open-telemetry/opentelemetry-go/pull/4804#issuecomment-1876601271) I decided to follow @MrAlias recommendation: 4f720aeb